src/proto/message.rs

```diff
@@ -65,7 +66,11 @@ impl Serialize for ExtensionContents {
     where
         S: serde::Serializer,
     {
-        serializer.serialize_bytes(&self.0)
+        let mut seq = serializer.serialize_tuple(self.0.len())?;
```

Comment:
Using a tuple is ... not great.
I don't really like it, but it's the only escape hatch I found.

This really comes down to the fact the protocol would expect:
```
struct SessionBind {
    pub extension_name: String,
    pub host_key: SshKey,
    pub session_id: Vec<u8>,
    pub signature: Signature,
    pub is_forwarding: bool,
}
```
along with `extension_name` being the hard coded value.

I think I like the idea of `ExtensionContents` though.

Anyway, I don't really have a good opinion on what's the best route to use.

Reply:
Yeah, it's kind of bad that the extension value is just data with any kind of shape and we'll be always chasing whatever SSH thinks of.

(My initial implementation just passed raw data but this was also suboptimal since the client had to parse it).

:thinking:

Reply:
Okay, I'll close this thread. I've created a ticket for more thorough [end-to-end tests with OpenSSH](https://github.com/wiktor-k/ssh-agent-lib/issues/12) which should catch any glaring issues that we may introduce.

I'll transplant them from another project in my free time (haha).